src/java.base/linux/classes/sun/nio/ch/FileDispatcherImpl.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2000, 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package sun.nio.ch;
+
+import java.io.FileDescriptor;
+
+class FileDispatcherImpl extends UnixFileDispatcherImpl {
+    FileDispatcherImpl() {
+        super();
+    }
+
+    int maxDirectTransferSize() {
+        return 0x7ffff000; // 2,147,479,552 maximum for sendfile()
+    }
+
+    long transferTo(FileDescriptor src, long position, long count,
+                    FileDescriptor dst, boolean append) {
+        return transferTo0(src, position, count, dst, append);
+    }
+
+    long transferFrom(FileDescriptor src, FileDescriptor dst,
+                      long position, long count, boolean append) {
+        return transferFrom0(src, dst, position, count, append);
+    }
+
+    // -- Native methods --
+
+    static native long transferTo0(FileDescriptor src, long position,
```

Comment:
can we make these native methods(transferTo0, transferFrom0 etc) to private access specifier,  these methods are not called from outside ?.

Reply:
This comment applies to a lot of code. See `UnixFileDispatcherImpl` for example. I prefer not to change it at this time.